### PR TITLE
Customizations, Troubleshooting, JSON Input Framework example

### PR DESCRIPTION
**Pull request by awelzel**

### CODE REVIEW

File: customizations.rst [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+.. _Apache Kafka: https://kafka.apache.org/
+.. _zeek-kafka: https://github.com/SeisoLLC/zeek-kafka/
+
+Log Rotation
+============
+
+JSON Streaming Logs
+-------------------
+
+The externally maintained `json-streaming-logs`_ package tailors Zeek
```
Comment (timwoj) [inline customizations.rst:100]:
```suggestion
The externally-maintained `json-streaming-logs`_ package tailors Zeek
```

File: customizations.rst [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+
+Profiling and Debugging
+=======================
+
+jemalloc profiling
+------------------
+
+For investigation of memory leaks or state-growth issues within Zeek,
+jemalloc's profiling is invaluable. A package providing a bit of help to
```
Comment (timwoj) [inline customizations.rst:118]:
```suggestion
jemalloc's profiling is invaluable. A package providing a bit of help for configuring jemalloc support
```

**Review by timwoj — CHANGES_REQUESTED** [review]

File: troubleshooting.rst [hunk 3]
```diff
@@ -0,0 +1,390 @@
+===============
+Troubleshooting
+===============
```
Comment (awelzel) [inline troubleshooting.rst:3]: @JustinAzoff  - not sure how easy this is for you to read, but if something comes to mind that could be generally valuable please state.

There hasn't been a place where we document basic `perf` or `jemalloc` usage/debugging and maybe Zeek docs is a bit of an odd place, but it's better than no place and tribe knowledge.

File: frameworks/input.rst [hunk 4]
```diff
@@ -442,6 +446,35 @@ returned line-by-line as strings; it can, for example, be used to read
...
+   :tab-width: 4
+
+.. literalinclude:: input_json_1.zeek
+   :caption: Loading denylist.jsonl, converting to Zeek types, populating a table.
+   :language: zeek
+   :linenos:
+   :tab-width: 4
+
+If your input data is already, or can be easily converted into JSON Lines format,
+above approach can be used to load it into Zeek.
```
Comment (timwoj) [inline frameworks/input.rst:476]:
```suggestion
If your input data is already in, or can be easily converted into, JSON Lines format
the above approach can be used to load it into Zeek.
```

File: troubleshooting.rst [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+Memory Leaks and State Growth
+=============================
+
+When memory of any Zeek process continuously grows in production or testing
+settings, there might be a memory leak in Zeek's C++ core or it might be
+scripting state growth. Examples of the latter are a global table that is
+populated but elements aren't expired or removed. Containers attached to
+connections may also cause unbounded state growth when these connections
+are long lived.
```
Comment (timwoj) [inline troubleshooting.rst:21]:
```suggestion
are long-lived.
```

File: troubleshooting.rst [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+
+For such issues, using jemalloc's memory profiling can be invaluable. A good
+introduction to this topic is `Justin Azhoff's profiling presentation`_ (`slides`_).
```
Comment (timwoj) [inline troubleshooting.rst:24]:
```suggestion
introduction to this topic is `Justin Azoff's profiling presentation`_ (`slides`_).
```

File: troubleshooting.rst [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+profiling enabled. Some Linux distributions provide a libjemalloc package that
+is configured this way. For example, Debian on the amd64 architecture
+has it enabled while Fedora 38 does not. You're advised to verify the
+``config.prof`` line in the jemalloc stats output as shown below.
+
+.. note::
+
+   If your distribution does not provide a suitable libjemalloc package,
+   building jemalloc from source configured with the required options
+   reasonably easy.
```
Comment (timwoj) [inline troubleshooting.rst:40]:
```suggestion
   is reasonably easy.
```

File: troubleshooting.rst [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+   config.prof_libunwind: false
+   ...
+
+
+If there is no output or ``config.prof`` says ``false``, verify your Zeek
+and libjemalloc setup.
+
+.. note::
+
+   Neither ``LD_PRELOAD`` nor ``MALLOC_CONF`` work with setuid or setcap
```
Comment (timwoj) [inline troubleshooting.rst:79]:
```suggestion
   Neither ``LD_PRELOAD`` nor ``MALLOC_CONF`` work with a setuid or setcap
```

File: troubleshooting.rst [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+   ``zeek`` binary and you might need to run as root or another privileged
+   enough user instead.
```
Comment (timwoj) [inline troubleshooting.rst:81]:
```suggestion
   user instead.
```

File: troubleshooting.rst [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+ZeekControl Integration
+~~~~~~~~~~~~~~~~~~~~~~~
+
+When working in a ZeekControl based environment, the `zeek-jemalloc`_ plugin
+can help with setting up the required environment variables. The ``.heap``
+files will be located in a worker's individual spool directory and can be
+processed with the ``jeprof`` utility as shown above.
+
+.. _zeek-jemalloc: https://github.com/JustinAzoff/zeek-jemalloc-profiling/tree/master
+.. _justin azhoff's profiling presentation: https://www.youtube.com/watch?v=gWSXbqxnJfs
```
Comment (timwoj) [inline troubleshooting.rst:138]:
```suggestion
.. _justin azoff's profiling presentation: https://www.youtube.com/watch?v=gWSXbqxnJfs
```

File: troubleshooting.rst [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+   }
+
+prof.log
+--------
+
+The ``prof.log`` provides aggregated information about Zeek's runtime status
+in a fairly non-structured text format.
+Likely future metrics will be added through the Telemetry framework mentioned
+above, but as of now it does contain information about queue sizes within
+the threading subsystem and other details that are not yet exposed otherwise.
```
Comment (timwoj) [inline troubleshooting.rst:355]: This sounds like something that a ticket could be opened for as a future enhancement.

**Review by timwoj — CHANGES_REQUESTED** [review]

File: troubleshooting.rst (hunk 6 above)
Comment (awelzel) [reply troubleshooting.rst:24]: Someday I'll get his name right. Looking at the github handle might have helped...

**Review by timwoj — APPROVED** [review]

**Pull request merged**